scripts/monitoring/cron-send-docker-metrics.py

```diff
@@ -46,8 +50,54 @@
         }
 
     zs.add_zabbix_keys(keys)
-
     print "Sending these metrics:"
     print json.dumps(keys, indent=4)
     zs.send_metrics()
     print "\nDone.\n"
+    #start auto-heal
+
+    if int(du_dds.data_space_percent_available) < 90:
+        print 'Docker has less than 90% storage avaiable. Attempting to clean up space.'
+        print '***********************************'
+        #clean the exited containers
+        for container in exited_containers:
+            print '*******start cleaning**************'
+            print 'the container id is :', container['Id']
+            #get the info
+            exited_container = cli.containers(all=True, filters={'id':container['Id']})
+            #print Deadcontainer
+            if len(exited_container) == 1:
+                print 'the status of the container is :', exited_container[0]['Status']
+                if exited_container[0]['Status'].find('Exited') != -1:
+                    #do the remove step
+                    print 'status confirmed :', exited_container[0]['Status']
+                    try:
+                        cli.remove_container(container=container['Id'])
+                        print 'done this container'
+                    except Exception, e:
+                        print 'something wrong during the remote of the container'
+            else:
+                print 'container not exist'
+        #clean the dead containers
+        for container in dead_containers:
+            print '*******start cleaning**************'
+            print 'the container id is :', container['Id']
+            #get the info
+            dead_container = cli.containers(all=True, filters={'id':container['Id']})
+            #print Deadcontainer
+            if len(dead_container) == 1:
+                print 'the status of the container is :', dead_container[0]['Status']
+                if dead_container[0]['Status'].find('Dead') != -1:
+                    #do the remove step
+                    print 'status confirmed :', dead_container[0]['Status']
+                    try:
+                        cli.remove_container(container=container['Id'])
+                        print 'done this container'
+                    except Exception, e:
+                        print 'something wrong during the remote of the container'
```

Comment:
It seems to me that the the handling of dead containers and exited containers is identical except for the status of 'Dead' or 'Exited'.

I'd put these in a function. Something like this (UNTESTED):
```
def remove_container(self, status):
    # insert the common code here

# ... snip ...

for container in dead_containers:
    self.remove_container('Exited')

# ... snip ...

for container in dead_containers:
    self.remove_container('Dead')
```

Always try to apply the DRY principle:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself